src/main/ruby/truffleruby/core/rational.rb

```diff
@@ -342,25 +346,94 @@ def self.convert(num, den)
     when Integer
       # nothing
     when String
-      num = String::Rationalizer.new(num).strict_convert
+      if exception
+        num = String::Rationalizer.new(num).strict_convert
+      else
+        num = String::Rationalizer.new(num).convert(nil)
+        return nil if Primitive.nil?(num)
+      end
```

Comment:
`convert(nil)` seems to me slightly better than `internal_convert()` that looks like a "not-public" method.